app/src/main/java/com/ordresot/playlistmaker/TrackAdapter.kt

```diff
@@ -1,14 +1,15 @@
 package com.ordresot.playlistmaker
 
 import android.view.LayoutInflater
+import android.view.View
 import android.view.ViewGroup
 import androidx.recyclerview.widget.RecyclerView
 
-class TrackAdapter(): RecyclerView.Adapter<TrackViewHolder>() {
+class TrackAdapter(private var trackList: ArrayList<Track>, private val trackOnClickListener: (Track) -> Unit): RecyclerView.Adapter<TrackViewHolder>() {
 
-    private var trackList = ArrayList<Track>()
-    fun setTrackList(trackList: ArrayList<Track>){
-        this.trackList = trackList
+    fun updateData(newTrackList: ArrayList<Track>){
```

Comment:
```suggestion
    fun updateData(newTrackList: List<Track>){
```
можно не завязываться на конкретную имплементацию списка